cdk/lib/support-workers.ts

```diff
@@ -232,5 +242,304 @@ export class SupportWorkers extends GuStack {
+      threshold: 1,
+    });
+
+    new GuAlarm(this, "TimeoutAlarm", {
+      app,
+      actionsEnabled: isProd,
+      snsTopicName: `alarms-handler-topic-${this.stage}`,
+      alarmName: `Support Workers Timeout in ${this.stage} (Recurring Contributions or Subscriptions Checkout)`,
+      alarmDescription: `There was a timeout whilst setting up recurring payments after the user attempted to complete a checkout process. Check https://eu-west-1.console.aws.amazon.com/states/home?region=eu-west-1#/statemachines/view/arn:aws:states:eu-west-1:865473395570:stateMachine:support-workers-${this.stage}?statusFilter=TIMED_OUT`,
+      metric: new Metric({
+        metricName: "ExecutionsTimedOut",
+        namespace: "AWS/States",
+        dimensionsMap: {
+          StateMachineArn: stateMachine.stateMachineArn,
+        },
+        statistic: "Sum",
+        period: Duration.seconds(60),
+      }),
+      comparisonOperator: ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+      evaluationPeriods: 1,
+      threshold: 1,
+    });
+
+    // This query is useful to check what a reasonable interval is for each product if you
+    // want the alarm to go off less than once a month.  Update the date range to get more current figures.
+    //
+    //  with timegaps as(
+    //  select
+    //  product,
+    //  if (print_options.product is null, null, if(print_options.product = 'GUARDIAN_WEEKLY', 'WEEKLY', 'NEWSPAPER')) print_product,
+    //  payment_provider,
+    //  e.event_timestamp,
+    //  (lag(e.event_timestamp) over (partition by payment_provider, product order by event_timestamp)) as last_acquisition,
+    //  from `datalake.fact_acquisition_event` e
+    //  where 1=1
+    //  and date(e.event_timestamp) >= date'2024-03-01' -- keep it after 1st March 2024 as the outage caused a S+ gap
+    //  and payment_provider in ('STRIPE', 'GOCARDLESS', 'PAYPAL')
+    //  and product in ('RECURRING_CONTRIBUTION', 'PRINT_SUBSCRIPTION', 'SUPPORTER_PLUS')
+    //  )
+    //  select
+    //  product,
+    //  print_product,
+    //  payment_provider,
+    //  max(timestamp_diff(event_timestamp, last_acquisition, HOUR)) max_diff,
+    //  from timegaps
+    //  group by 1,2,3
+    //  order by 1,2,3
+    new GuAlarm(this, "NoPaypalContributionsAlarm", {
+      app,
+      actionsEnabled: isProd,
+      snsTopicName: `alarms-handler-topic-${this.stage}`,
+      alarmName: `support-workers ${this.stage} No successful recurring paypal contributions recently`,
+      metric: new Metric({
+        metricName: "PaymentSuccess",
+        namespace: "support-frontend",
+        dimensionsMap: {
+          PaymentProvider: "PayPal",
+          ProductType: "Contribution",
+          Stage: this.stage,
+        },
+        statistic: "Sum",
+        period: Duration.seconds(3600),
+      }),
+      comparisonOperator: ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
+      evaluationPeriods: 4,
+      treatMissingData: TreatMissingData.BREACHING,
```

Comment:
I assume this is how the alarm was already set up rather than a change you made, but actually this will take a fair bit longer than 3600s * 4 (4 hours) to go off, as when there is missing data it looks back to find more data earlier.  The usual solution is to use a calculated metric and use `FILL(m1, 0)` as the expression, where `m1` is the actual PaymentSuccess metric.

Reply:
see example on line 453 (although that also SUMs several metric which is not needed here)